libs/base/Data/List/Quantifiers.idr

```diff
@@ -0,0 +1,84 @@
+module Data.List.Quantifiers
+
+import Data.List
+import Data.List.Elem
+
+||| A proof that some element of a list satisfies some property
+|||
+||| @ p the property to be satisfied
+public export
+data Any : (0 p : a -> Type) -> List a -> Type where
+  ||| A proof that the satisfying element is the first one in the `List`
+  Here  : {0 xs : List a} -> p x -> Any p (x :: xs)
+  ||| A proof that the satisfying element is in the tail of the `List`
+  There : {0 xs : List a} -> Any p xs -> Any p (x :: xs)
+
+export
+Uninhabited (Any p Nil) where
+  uninhabited (Here _) impossible
+  uninhabited (There _) impossible
+
+||| Eliminator for `Any`
+export
+anyElim : (Any p xs -> b) -> (p x -> b) -> Any p (x :: xs) -> b
+anyElim _ f (Here p)  = f p
```

Comment:
That looks more like a `case` than an `elim` as we are not calling
ourselves recursively in the `There` case (and the target type is
not dependent).

Reply:
Yeah, this was just copied from Idris1, and I'm not even sure what's the use case for it, maybe let's just remove it?

Reply:
Me neither... if it turns out anyone needs it I'm sure we'll welcome the PR!